src/javascripts/components/cookie-functions.js

```diff
@@ -157,14 +159,19 @@ export function resetCookies () {
       Analytics()
     }
 
-    // Delete cookies of that type if consent is false
     if (!options[cookieType]) {
-      for (var cookie in COOKIE_CATEGORIES) {
-        if (COOKIE_CATEGORIES[cookie] === cookieType) {
-          // Delete cookie
-          Cookie(cookie, null)
+      // Fetch that category lookup function
+      var cookiesInCategory = COOKIE_CATEGORIES[cookieType]
+      var cookies = document.cookie.split(';')
+
+      cookies.forEach(function (cookie) {
+        var cookieName = cookie.split('=')[0].replace(' ', '')
+
+        // Delete all cookies in that category
+        if (cookiesInCategory.indexOf(cookieName) >= 0) {
```

Comment:
I feel like the code getting the existing cookies is redundant; surely deleting a cookie that isn't already set should be fine, right? Or did you find problems with that in testing?

Reply:
That's a really good shout, for some reason it didn't occur to me 👍 I don't think that would cause any problems... I'll do some testing to double check

Reply:
@lfdebrux I've tweaked this now so we no longer fetch the existing cookies ([commit](https://github.com/alphagov/govuk-design-system/pull/1791/commits/7318988fe0ca0045746ee2528de5713d76b8d1d2#diff-54c970228ba5fe3657fcdee2af4308d31ce1c41eff42a21b9405e919664cd25fR164))